packages/examples/fortune-v2/job-launcher/server/src/modules/payment/payment.service.ts

```diff
@@ -0,0 +1,159 @@
+import { Injectable, Logger, NotFoundException } from "@nestjs/common";
+import { ConfigService } from "@nestjs/config";
+import Stripe from "stripe";
+import { BigNumber, Signer } from 'ethers';
+import { ErrorPayment } from "../../common/constants/errors";
+import { PaymentRepository } from "./payment.repository";
+import { CurrencyService } from "./currency.service";
+import { PaymentCryptoCreateDto, PaymentFiatConfirmDto, PaymentFiatCreateDto } from "./payment.dto";
+import { Currency, PaymentFiatMethodType, PaymentSource, PaymentStatus, PaymentType, TokenId } from "../../common/enums/payment";
+import { EthersSigner, InjectSignerProvider } from "nestjs-ethers";
+import { Wallet } from "@ethersproject/wallet";
+import { TX_CONFIRMATION_TRESHOLD } from "../../common/constants";
+
+@Injectable()
+export class PaymentService {
+  private readonly logger = new Logger(PaymentService.name);
+  private stripe: Stripe;
+  private endpointSecrete: string;
+
+  // TODO: add crypto variant
+  constructor(
+    @InjectSignerProvider()
+    private readonly ethersSigner: EthersSigner,
+    private readonly paymentRepository: PaymentRepository,
+    private readonly currencyService: CurrencyService,
+    private configService: ConfigService,
+  ) {
+    this.stripe = new Stripe(this.configService.get("STRIPE_SECRET_KEY", "secrete-key"), {
+      apiVersion: this.configService.get("STRIPE_API_VERSION", "2022-11-15"),
+      appInfo: {
+        name: this.configService.get("NAME", "Fortune"),
+        version: this.configService.get("VERSION"),
+        url: this.configService.get("STRIPE_APP_INFO_URL"),
+      },
+    });
+    this.endpointSecrete = this.configService.get("STRIPE_ENDPOINT_SECRETE", "secrete-key");
+  }
+
+  public async createCustomer(email: string) {
+    const customer = await this.stripe.customers.create({
+      email,
+    });
+
+    if (!customer) {
+      this.logger.log(ErrorPayment.CustomerNotFound, PaymentService.name);
+      throw new NotFoundException(ErrorPayment.CustomerNotFound);
+    }
+
+    return customer.id;
+  }
+
+  public async createFiatPayment(customerId: string, dto: PaymentFiatCreateDto) {
+    const { amount, currency } = dto;
+    const paymentMethodOptions = {};
+
+    const params: Stripe.PaymentIntentCreateParams = {
+      payment_method_types: [PaymentFiatMethodType.CARD],
+      amount: amount * 100,
+      currency: currency,
+    };
+
+    params.confirm = true;
+    params.customer = customerId;
+
+    if (paymentMethodOptions) {
+      params.payment_method_options = paymentMethodOptions;
+    }
+
+    const paymentIntent = await this.stripe.paymentIntents.create(params);
+
+    return {
+      clientSecret: paymentIntent.client_secret
+    };
+  }
+
+  public async confirmFiatPayment(userId: number, dto: PaymentFiatConfirmDto): Promise<boolean> {
+    try {
+      const paymentData = await this.getPayment(dto.paymentId);
+
+      if (paymentData?.status?.toUpperCase() !== PaymentStatus.SUCCEEDED) {
```

Comment:
Can a FAILED payment turn SUCCEEDED?

Reply:
It depends on reason, I think we can handle this cases later, for now, I propose to keep it simple)